LuaMenu/widgets/chobby/components/console.lua

```diff
@@ -185,8 +233,16 @@ end
 function Console:SendMessage()
 	if self.ebInputText.text ~= "" then
 		message = self.ebInputText.text
-		if self.listener then
-			self.listener(message)
+		if self.sendMessageActions then
+			if message:starts("/me ") then
+				self.sendMessageActions.ircStyle(message:sub(5))
+			else
+				self.sendMessageActions.default(message)
+			end
+		end
+		-- If the message is different than the last sent message, save the message for retrieval by uparrow.
+		if self.sentMessages[2] ~= self.ebInputText.text then
+			table.insert(self.sentMessages, 2, self.ebInputText.text)
 		end
```

Comment:
I think you want to reset the sentMessageIndex here as well, maybe even when tab is pressed

Reply:
Yes, fixed for both.